### PR TITLE
Add Unit Tests

### PR DESCRIPTION
**Pull request by Wolemercy**

### CODE REVIEW

File: TodoApi.Tests/TodoApiTests.cs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+using TodoApiWithControllers.Models;
+using TodoApiWithControllers.Services;
+using TodoApiWithControllers.Controllers;
+
+public class UnitTest1
+{
+    [Fact]
+    public async void PostTodoItem_CreatesATodoItem()
+    {
+        var mockTodoItemDTO = new Mock<TodoItemDTO>();
```
Comment (allengblack) [inline TodoApi.Tests/TodoApiTests.cs]: You can use [this](https://github.com/bchavez/Bogus) to generate fake data.

File: TodoApi.Tests/TodoApiTests.cs [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        var mockTodoService = new Mock<ITodoService>();
+        mockTodoService.Setup(svc => svc.CreateTodoItem(It.IsAny<TodoItemDTO>()))
+            .ReturnsAsync(GetTodos()[0]);
+
+        var controller = new TodoItemsController(mockTodoService.Object);
+
+        var result = await controller.PostTodoItem(mockTodoItemDTO.Object);
```
Comment (allengblack) [inline TodoApi.Tests/TodoApiTests.cs]: You generally want to assert the results, too. 
https://fluentassertions.com/introduction

File: TodoApi.Tests/TodoApiTests.cs (hunk 1 above)
Comment (Wolemercy) [reply TodoApi.Tests/TodoApiTests.cs]: Aye, I'll check that out. Thank you

File: TodoApi.Tests/TodoApiTests.cs (hunk 2 above)
Comment (Wolemercy) [reply TodoApi.Tests/TodoApiTests.cs]: Haha, I tried, but nothing seemed to work. I'll take a look at the link, thanks

File: TodoApi.Tests/TodoServiceTests.cs [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+    {
+        var expectedResult = new Faker<TodoItem>()
+            .RuleFor(u => u.Id, f => f.Random.Long(0, 0))
+            .RuleFor(u => u.Name, f => f.Hacker.Verb())
+            .RuleFor(u => u.IsCompleted, f => f.Random.Bool())
+            .Generate(3);
+
+        var data = expectedResult.AsQueryable();
+
+        var mockSet = new Mock<DbSet<TodoItem>>();
```
Comment (allengblack) [inline TodoApi.Tests/TodoServiceTests.cs:106]: var context = new Mock<TodoContext>();
context.Setup(c => c.Todo).Returns(data);

**Review by allengblack — APPROVED** [review]

**Pull request merged**